Makefile

```diff
@@ -1,31 +1,34 @@
+CONTRACTS := $(wildcard ./contracts/*/.)
+CIRCUITS := $(wildcard ./circuits/*/.)
+
 help: ## Display this help screen
 	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-15s\033[0m %s\n", $$1, $$2}'
 
-wasm: ## Generate the WASM for the contract given (e.g. make wasm for=transfer)
-	@cargo rustc \
-		--manifest-path=contracts/$(for)/Cargo.toml \
-		--release \
-		--target wasm32-unknown-unknown \
-		-- -C link-args=-s
-
-contracts: ## Generate the WASM for all the contracts & test them
-	$(MAKE) -C ./contracts/transfer/ test
-	$(MAKE) -C ./contracts/bid/ test
+abi: ## Build the ABI and test it
+	$(MAKE) -C ./rusk-abi test
 
 keys: ## Create the keys for the circuits
-	cd rusk && cargo test --release -vv -- --nocapture
+	$(MAKE) -C ./rusk keys
+
+wasm: ## Generate the WASM for all the contracts
+	for dir in $(CONTRACTS); do \
+        $(MAKE) -C $$dir $@ ; \
+    done
```

Comment:
We can't go that way because we have 3 different tasks for the same directory's name: wasm, circuits and tests

And ofc, it doesn't make sense to adapt the dirs name to any needs of the Makefile

Reply:
This is not the case, it would be without `$(MAKECMDGOALS)` (like `rusk-abi` is, since it doesn't have more than one command), but the code above is the same as `rusk-abi`'s one, with the addition of `$(MAKECMDGOALS)` to propagate the specific recipe down to the sub-folder's `Makefile`.

Of course as I mentioned the sub-folder's Makefile should have the same structure as the parent's one, but I don't think that would be a problem: to be clear, the contract's Makefile should have both `wasm` and `test` recipe, as the parent does, and in the `test` of the contract's we should call the contract's `wasm` recipe before running `test`, if it's supposed to be a stand alone Makefile.
Otherwise if it's not supposed to be also a stand-alone Makefile, as `rusk-abi`, should have only one recipe and the tests should be in the parent's one only.

If this is still unclear I can create a PR on top of yours with the above approach.